### ISSUE TITLE
ERROR Subscription Liveness timeout every 5 minutes

### ISSUE THREAD

**Issue by bigbangus** [home-assistant-libs/python-matter-server#395]
I'm using an Aqara E1 hub with an Aqara U100 door lock.

The door lock is connected to HA Core docker using the matter-server docker in Unraid 6.12.4.

After setting up in homeassistant, it works, but it resets every 5 minutes. Is there a bug in matter-server causing this error?

```
2023-09-13 19:33:00 UnraidNAS PersistentStorage[1] INFO Committing...
2023-09-13 19:33:00 UnraidNAS root[1] INFO Re-subscription succeeded!
2023-09-13 19:33:00 UnraidNAS matter_server.server.device_controller.[node 1][1] INFO Re-Subscription succeeded
2023-09-13 19:39:38 UnraidNAS chip.DMG[1] ERROR Subscription Liveness timeout with SubscriptionID = 0x33f6410c, Peer = 01:0000000000000001
2023-09-13 19:39:38 UnraidNAS matter_server.server.device_controller.[node 1][1] INFO Previous subscription failed with Error: 50, re-subscribing in 0 ms...
2023-09-13 19:39:39 UnraidNAS PersistentStorage[1] INFO DeleteSdkKey: g/s/Uc8vwS2xf1tfddjms/WkJQ==
2023-09-13 19:39:39 UnraidNAS PersistentStorage[1] INFO Committing...
2023-09-13 19:39:39 UnraidNAS PersistentStorage[1] INFO SetSdkKey: f/1/s/0000000000000001 = b'\x150\x03\x10 u{\x08\xb4Nf\xb1"\x8f\x15H\xf9\x97q\xf30\x04 \x96\x8a\x08\xcca.R6\\)\xe5\x81h\xcd\x06\x1f\xa30\xd8,\x8c\x9fZ\x15\xfd\x1c)\xf5\xa8\x1f\xf7J0\x05\x0c\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x18'
2023-09-13 19:39:39 UnraidNAS PersistentStorage[1] INFO Committing...
2023-09-13 19:39:39 UnraidNAS PersistentStorage[1] INFO SetSdkKey: g/s/IHV7CLROZrEijxVI+Zdx8w== = b'\x15$\x01\x01$\x02\x01\x18'
2023-09-13 19:39:39 UnraidNAS PersistentStorage[1] INFO Committing...
2023-09-13 19:39:39 UnraidNAS root[1] INFO Re-subscription succeeded!
2023-09-13 19:39:39 UnraidNAS matter_server.server.device_controller.[node 1][1] INFO Re-Subscription succeeded
2023-09-13 19:44:43 UnraidNAS chip.DMG[1] ERROR Subscription Liveness timeout with SubscriptionID = 0x5625e1e5, Peer = 01:0000000000000001
2023-09-13 19:44:43 UnraidNAS matter_server.server.device_controller.[node 1][1] INFO Previous subscription failed with Error: 50, re-subscribing in 0 ms...
2023-09-13 19:44:44 UnraidNAS PersistentStorage[1] INFO DeleteSdkKey: g/s/IHV7CLROZrEijxVI+Zdx8w==
2023-09-13 19:44:44 UnraidNAS PersistentStorage[1] INFO Committing...
2023-09-13 19:44:44 UnraidNAS PersistentStorage[1] INFO SetSdkKey: f/1/s/0000000000000001 = b'\x150\x03\x10$\xc2?%\xa4\x9f\xe7yu/\r\x94\xd5\x9eH%0\x04 \x96\x8a\x08\xcca.R6\\)\xe5\x81h\xcd\x06\x1f\xa30\xd8,\x8c\x9fZ\x15\xfd\x1c)\xf5\xa8\x1f\xf7J0\x05\x0c\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x18'
2023-09-13 19:44:44 UnraidNAS PersistentStorage[1] INFO Committing...
2023-09-13 19:44:44 UnraidNAS PersistentStorage[1] INFO SetSdkKey: g/s/JMI/JaSf53l1Lw2U1Z5IJQ== = b'\x15$\x01\x01$\x02\x01\x18'
2023-09-13 19:44:44 UnraidNAS PersistentStorage[1] INFO Committing...
2023-09-13 19:44:45 UnraidNAS root[1] INFO Re-subscription succeeded!
2023-09-13 19:44:45 UnraidNAS matter_server.server.device_controller.[node 1][1] INFO Re-Subscription succeeded
2023-09-13 19:49:49 UnraidNAS chip.DMG[1] ERROR Subscription Liveness timeout with SubscriptionID = 0x44513c55, Peer = 01:0000000000000001
2023-09-13 19:49:49 UnraidNAS matter_server.server.device_controller.[node 1][1] INFO Previous subscription failed with Error: 50, re-subscribing in 0 ms...
2023-09-13 19:49:50 UnraidNAS PersistentStorage[1] INFO DeleteSdkKey: g/s/JMI/JaSf53l1Lw2U1Z5IJQ==
2023-09-13 19:49:50 UnraidNAS PersistentStorage[1] INFO Committing...
2023-09-13 19:49:50 UnraidNAS PersistentStorage[1] INFO SetSdkKey: f/1/s/0000000000000001 = b'\x150\x03\x10\xc4\x99\x89\xa7\xe4p\xc1E\xf1l\x17\xe9\xf5\xaaA\x140\x04 \x96\x8a\x08\xcca.R6\\)\xe5\x81h\xcd\x06\x1f\xa30\xd8,\x8c\x9fZ\x15\xfd\x1c)\xf5\xa8\x1f\xf7J0\x05\x0c\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x18'
2023-09-13 19:49:50 UnraidNAS PersistentStorage[1] INFO Committing...
2023-09-13 19:49:50 UnraidNAS PersistentStorage[1] INFO SetSdkKey: g/s/xJmJp+RwwUXxbBfp9apBFA== = b'\x15$\x01\x01$\x02\x01\x18'
2023-09-13 19:49:50 UnraidNAS PersistentStorage[1] INFO Committing...
2023-09-13 19:49:50 UnraidNAS root[1] INFO Re-subscription succeeded!
2023-09-13 19:49:50 UnraidNAS matter_server.server.device_controller.[node 1][1] INFO Re-Subscription succeeded
```

![image](https://github.com/home-assistant-libs/python-matter-server/assets/91270101/2279c733-56c2-4660-8bbb-58c127745f73)

**Comment by farmio** [comment]
Same issue here with a container install using a HUE Hub. Additionally I see some
> ERROR [139829803208592] Client exceeded max pending messages: 512

messages. See [matter-server-log.txt](https://github.com/home-assistant-libs/python-matter-server/files/12682328/matter-server-log.txt)

I can provide more logs or test things if needed.

**Comment by marcelveldt** [comment]
First of all, use the Home Assistant issue tracker for issues with Home Assistant, this repository is for the matter server/library only. 

This looks like some connection issues as the subscription times out. As running matter server yourself in docker is highly experimental, please see if you can reproduce this same issue when using Home Assistant OS. For example run it in a VM or something. 

Communication issues like this we see a lot due to Matter completely relying on (local) IPv6 multicast (mdns) for communication and network gear not completely ready up to that task or just filtering it partly out. That is also true for some linux kernels, hence our request to test this with HAOS as that is the platform we ensured it will work with.

**Comment by marcelveldt** [comment]
There is no such thing as "sharing mode". A Matter device can be commissioned to multiple fabrics (controllers) at the same time. So sharing it from Apple Home just means that Apple Home will tell the device to open a commissioning window for the other controller. Each controller interacts with the device completely independently.

As for using your using own OTBR with HA; that is HIGHLY experimental, especially when you run it yourself in docker as you need special kernel patches to resolve routing issues. At this time we do not have the capacity to support you with that setup, I'm sorry.

My advice to everyone at this point is to use existing, established border routers, such as the Apple Homepod Mini. That works rock stable at this point, also to use in HA. In the meanwhile we're making progress in using the HA Yellow or SkyConnect a stand-alone Thread border router including some descent UI options flow.

**Comment by marcelveldt** [comment]
Going to close here. Please use the HA core issue tracker to report issues with Matter devices in HA and be aware of the supported scenarios. So if you are using one of the setup methods described in the documentation and you run into this issue, please open a new issue in the HA core issue tracker, including the diagnostics dump so we can have a look at it. Thanks!

**Comment by bigbangus** [comment]
The issue was just fixed with 4.0.1 release for whatever reason!

As soon as I updated the docker from the previous version (I keep my dockers current), the HA status become stable. See picture below around 8PM.

For the record I'm on a pfSense router and Unifi switches and APs.

![image](https://github.com/home-assistant-libs/python-matter-server/assets/91270101/04c8dc95-7bb9-4c67-9d43-33302a032cbf)
